server/sockets/socketsChallengeArena.js

```diff
@@ -86,6 +91,73 @@ module.exports = function (io) {
         console.log('Client disconnected after having started a challenge', socket.id);
       }
     });
+
+    //----------Pair Socket Event Handlers ----------
+    socket.on('pair', function (github_handle) {
+      // if there aren't any open room, create a room and join it
+      if (pairOpenQ.length === 0) {
+        // create a room
+        pairRoomCounter++;
+        console.log('socketsChallenge line-98, Creating and joining new PAIR ROOM', roomCounter);
+        socket.join(String(roomCounter));
+        // add this room to the pairOpenQ
+        pairOpenQ.push({
+          name: roomCounter,
+          players: [github_handle],
+          socket_id: [socket.id]
+        });
+      // Otherwise, there is an open room, join that one
+      } else {
+        var existingRoom = pairOpenQ.shift();
+        // join the first existing room
+        console.log('server.js line-93, Joining existing room:', existingRoom.name);
+        socket.join(String(existingRoom.name));
+        // remove this room from the pairOpenQ and add to inProgressRooms
+        // find all players in the room and find a challenge neither player has seen
+        var otherPlayer = existingRoom.players[0];
+        challengeController.getChallengeMultiplayer({
+          body: {
+            player1_github_handle: otherPlayer,
+            player2_github_handle: github_handle,
+            type: 'pair'
+          }
+        }, function (challenge) {
+          if (challenge !== null) {
+            solutionController.initializeChallengeSolutions(otherPlayer, github_handle, challenge.id, 'pair');
+          } else {
+            challenge = {
+              id: null,
+              name: null,
+              prompt: '/*Sorry we ran out of problems! \nPlease exit and re-enter the room to try again*/'
+            }
+          }
+          io.to(String(existingRoom.name)).emit('pair_up', challenge);
+        });
+      }
+    });
+
+    socket.on('ready', function(data){
+      socket.to(findRoom(socket)).broadcast.emit('ready', data);
+    });
+
+    socket.on('start_pair', function(startData) {
+      solutionController.initializeChallengeSolutions(startData.opponent_github_handle, startData.user_github_handle, startData.challenge_id);
+      socket.to(findRoom(socket)).broadcast.emit('start_pair', data);
+    });
+
+    socket.on('we_done', function(data){
```

Comment:
'we_done' and 'we_won' are not necessary, we should be able to handle these in a similar manner to challenge_arena